code/game/gamemodes/malfunction/Malf_Modules.dm

```diff
@@ -656,35 +656,37 @@
 	mod_pick_name = "recam"
 	description = "Runs a network-wide diagnostic on the camera network, resetting focus and re-routing power to failed cameras. Can be used to repair up to 30 cameras."
 	cost = 10
-	one_purchase = TRUE
 	power_type = /datum/action/innate/ai/reactivate_cameras
 	unlock_text = "<span class='notice'>You deploy nanomachines to the cameranet.</span>"
 
 /datum/action/innate/ai/reactivate_cameras
 	name = "Reactivate Cameras"
 	desc = "Reactivates disabled cameras across the station; remaining uses can be used later."
 	button_icon_state = "reactivate_cameras"
-	uses = 30
+	uses = 10
 	auto_use_uses = FALSE
-	cooldown_period = 30
+	cooldown_period = 3 SECONDS
 
 /datum/action/innate/ai/reactivate_cameras/Activate()
-	var/fixed_cameras = 0
-	for(var/V in GLOB.cameranet.cameras)
+	var/mob/living/silicon/ai/user = usr
+	var/repaired_cameras = 0
+	if(!istype(user))
+		return
+	for(var/obj/machinery/camera/camera_to_repair in get_area(user.eyeobj)) // replace with the camera list on areas when that list actually works, the UIDs change right now so it (almost) always fails
```

Comment:
Is this a todo of sorts?

Reply:
yeah, but not really for me, just for anyone that actually fixes the list